src/axom/sidre/spio/IOManager.cpp

```diff
@@ -793,14 +796,18 @@ std::string IOManager::getFileNameForRank(const std::string& file_pattern,
 
   //If the root file was given as a path,
   //find the directory and add it to file_name
-  std::string curr;
-  std::string root_dir;
-  std::string slash = "/";
-  conduit::utils::rsplit_string(root_name, slash, curr, root_dir);
+  std::string delimiter = "/";
+  axom::Path root_path(root_name);
+  std::string root_dir = root_path.dirName();
+
+  if (!root_name.empty() && root_name[0] == delimiter[0])
+  {
+    root_dir = delimiter + root_dir;
+  }
```

Comment:
One limitation of axom::Path is that it does not preserve a leading '/' delimiter, which can make a meaningful difference when used as a file system path, so lines 803-806 restore this when necessary.

Reply:
I would argue that's a bug and not intentional.  Did you attempt to fix this in `axom::Path` and other things broke?

Reply:
I didn't touch `axom::Path`.  I could try to make a fix there before merging this PR.

Reply:
It is fixable in `axom::Path` and I got most of the way there this afternoon.  However, I think it should be a standalone change and not bloat the scope of this PR.  I'm inclined to create another issue and PR.  I could merge this one with the 4-line workaround, or let it wait for the other one.

Reply:
Issue #777 created regarding the leading delimiter in `axom::Path`.

Reply:
i'm okay with that

Reply:
I agree that this is a bug and that it can be resolved with the current workaround for this PR with a follow-up fix in `axom::Path`, as discussed in #777.